applicationinsights-react-js/src/ReactPlugin.ts

```diff
@@ -31,24 +37,27 @@ export default class ReactPlugin extends BaseTelemetryPlugin {
 
             _self.initialize = (config: IConfiguration & IConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?:ITelemetryPluginChain) => {
                 super.initialize(config, core, extensions, pluginChain);
-                _extensionConfig =
-                    config.extensionConfig && config.extensionConfig[_self.identifier]
-                        ? (config.extensionConfig[_self.identifier] as IReactExtensionConfig)
-                        : { history: null };
-        
-                arrForEach(extensions, ext => {
-                    const identifier = (ext as ITelemetryPlugin).identifier;
-                    if (identifier === 'ApplicationInsightsAnalytics') {
-                        _analyticsPlugin = (ext as any) as IAppInsights;
+
+                _self._addHook(onConfigChange(config, (details) => {
+                    let ctx = _self._getTelCtx();
+                    _extensionConfig = ctx.getExtCfg<IReactExtensionConfig>(this.identifier, defaultReactExtensionConfig);
+                    arrForEach(extensions, ext => {
```

Comment:
This can be replaced with the following now, as v3.x always has the getPlugin function.
```ts
let loadedPlugin = core.getPlugin("ApplicationInsightsAnalytics");
_analyticsPlugin = loadedPlugin.plugin;
```

Reply:
ie. same as the react-native